src/controllers/pagoPAProxyController.ts

```diff
@@ -31,10 +31,10 @@ export default class PagoPAProxyController {
     | IResponseErrorValidation
     | IResponseSuccessJson<PaymentRequestsGetResponse>
   > =>
-    withValidatedOrInternalError(
-      t.string.decode(req.params.rptId),
-      this.pagoPAProxyService.getPaymentInfo
-    );
+    withValidatedOrInternalError(t.string.decode(req.params.rptId), rptId => {
+      const isTest = String(req.query.test).toLowerCase() === "true";
```

Comment:
you should probably extract this check to a function to avoid repeating the logic

```ts
const parsePagopaTestParam = (testParam: unknown) => String(req.query.test).toLowerCase() === "true";
```

once you do that, you can shorten this code to:

```ts
withValidatedOrInternalError(t.string.decode(req.params.rptId), rptId =>
  this.pagoPAProxyService.getPaymentInfo(rptId, parsePagopaTestParam(req.query.test))
)

Reply:
Changed in 455eb59